packages/tpetra/core/src/Tpetra_Details_unpackCrsMatrixAndCombine_decl.hpp

```diff
@@ -195,33 +195,48 @@ unpackAndCombineWithOwningPIDsCount (
 ///
 /// Note: This method does the work previously done in unpackAndCombineWithOwningPIDsCount,
 /// namely, calculating the local number of nonzeros, and allocates CRS
-/// arrays of the correct sizes.
+/// arrays of the correct sizes
 
 template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
 void
 unpackAndCombineIntoCrsArrays (
     const CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> & sourceMatrix,
     const Kokkos::View<LocalOrdinal const *, 
           Kokkos::Device<typename Node::device_type::execution_space,
-                         Tpetra::Details::DefaultTypes::comm_buffer_memory_space<typename Node::device_type>>,
-          void, void>,
+                         Tpetra::Details::DefaultTypes::comm_buffer_memory_space<typename Node::device_type>>
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+          , void, void
+#endif
```

Comment:
Wouldn't it be possible to just delete `void, void` here and in all the subsequent occurrences?

Reply:
@cgcgcg no, it is needed for backward compatibility with the prior DualView interface, otherwise compilation will fail with a type mismatch

Reply:
@cgcgcg here is a PR for reference https://github.com/kokkos/kokkos/pull/6155 related to the issue that removing deleting the `void, void` template args would trigger with `KOKKOS_ENABLE_DEPRECATED_CODE_4`
When changes were made to the DualView interface (replacing the template args with variadic pack, i.e. "`DualView<class, class=void, class=void, class=void>` -> `DualView<class, class...>`") it broke codes, and needed to be reintroduced within `KOKKOS_ENABLE_DEPRECATED_CODE_4` for compatibility